app/Http/Requests/v1/AlterRoleRequest.php

```diff
@@ -14,8 +14,7 @@ class AlterRoleRequest extends FormRequest
      */
     public function authorize()
     {
-        //The body parameter in the request for role creation/edition must not be equal than the model for implicit binding issues when trying to edit (PUT)
-        $role = $this->route()->parameter('role');
+        $role = $this->route()->parameter('role_name');
```

Comment:
```suggestion
        $role = $this->route()->parameter('role');
```

Reply:
this role is the role on the url and this must be different to the super role.
if the role_name in the body request is equal to the super role it will be validate in the rules section